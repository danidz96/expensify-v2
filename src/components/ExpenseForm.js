import React, { Component } from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

moment.locale('es');
console.log(moment().format('LL'));

export class ExpenseForm extends Component {
	constructor(props) {
		super();

		this.state = {
			description: props.expense ? props.expense.description : '',
			amount: props.expense ? (props.expense.amount / 100).toString() : '',
			note: props.expense ? props.expense.note : '',
			createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
			focused: false,
			error: ''
		};
	}

	onDescriptionChange = (e) => {
		const description = e.target.value;
		this.setState({
			description
		});
	};

	onAmountChange = (e) => {
		const amount = e.target.value;
		if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
			this.setState({
				amount
			});
		}
	};

	onNoteChange = (e) => {
		const note = e.target.value;
		this.setState({
			note
		});
	};

	onDateChange = (createdAt) => {
		if (createdAt) {
			this.setState({
				createdAt
			});
		}
	};

	onFocusChange = ({ focused }) => {
		this.setState({
			focused
		});
	};

	onSubmit = (e) => {
		e.preventDefault();
		if (!this.state.description || !this.state.amount) {
			this.setState({ error: 'Please prove description and amount' });
		} else {
			this.setState({ error: '' });
			this.props.onSubmit({
				description: this.state.description,
				amount: parseFloat(this.state.amount, 10) * 100,
				createdAt: this.state.createdAt.valueOf(),
				note: this.state.note
			});
		}
	};

	render() {
		return (
			<div>
				{this.state.error && <p>{this.state.error}</p>}
				<form onSubmit={this.onSubmit}>
					<input
						type="text"
						placeholder="Description"
						autoFocus
						value={this.state.description}
						onChange={this.onDescriptionChange}
					/>
					<input type="text" placeholder="Amount" value={this.state.amount} onChange={this.onAmountChange} />
					<SingleDatePicker
						date={this.state.createdAt}
						onDateChange={this.onDateChange}
						focused={this.state.focused}
						onFocusChange={this.onFocusChange}
						id="form_datepicker"
						numberOfMonths={1}
						isOutsideRange={() => false}
					/>
					<textarea
						placeholder="Add a note for your expense (optional)"
						value={this.state.note}
						onChange={this.onNoteChange}
					/>
					<button>Add Expense</button>
				</form>
			</div>
		);
	}
}

export default ExpenseForm;
