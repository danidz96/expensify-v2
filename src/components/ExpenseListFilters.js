import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';
import { DateRangePicker } from 'react-dates';

class ExpenseListFilters extends Component {
	constructor(props) {
		super(props);

		this.state = {
			calendarFocused: null
		};
	}

	onDatesChange = ({ startDate, endDate }) => {
		this.props.dispatch(setStartDate(startDate));
		this.props.dispatch(setEndDate(endDate));
	};

	onFocusChange = (calendarFocused) => {
		this.setState({ calendarFocused });
	};

	render() {
		return (
			<div>
				<input
					type="text"
					value={this.props.filters.text}
					onChange={(e) => {
						this.props.dispatch(setTextFilter(e.target.value));
					}}
				/>
				<select
					value={this.props.filters.sortBy}
					onChange={(e) => {
						if (e.target.value === 'date') {
							this.props.dispatch(sortByDate());
						} else if (e.target.value === 'amount') {
							this.props.dispatch(sortByAmount());
						}
					}}
				>
					<option value="date">Date</option>
					<option value="amount">Amount</option>
				</select>
				<DateRangePicker
					startDate={this.props.filters.startDate}
					startDateId="filter_start_date"
					endDate={this.props.filters.endDate}
					endDateId="filter_end_date"
					onDatesChange={this.onDatesChange}
					focusedInput={this.state.calendarFocused}
					onFocusChange={this.onFocusChange}
					isOutsideRange={() => false}
					numberOfMonths={1}
					showClearDates={true}
				/>
			</div>
		);
	}
}

const mapStateToProps = (state) => {
	return {
		filters: state.filters
	};
};

export default connect(mapStateToProps)(ExpenseListFilters);
