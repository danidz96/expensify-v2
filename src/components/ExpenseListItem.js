import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

export const ExpenseListItem = ({ id, description, amount, createdAt }) => (
	<div>
		<Link to={`/edit/${id}`}>
			<h3>{description}</h3>
		</Link>
		<p>
			{numeral(amount / 100).format('0,0.[00]$')}
			-
			{moment(createdAt).format('Do MMMM, YYYY')}
		</p>
	</div>
);

//Without destructuring

// const ExpenseListItem = (props) => {
//   const { description, amount, createdAt } = props.expense;
//   return (
//     <div>
//       <h3>{description}</h3>
//       <p>{amount} - {createdAt}</p>
//     </div>
//   )
// }

export default ExpenseListItem;
