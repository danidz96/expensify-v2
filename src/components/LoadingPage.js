import React from 'react';
import { ImpulseSpinner } from 'react-spinners-kit';

const LoadingPage = () => (
	<div className="loader">
		<ImpulseSpinner size={80} frontColor="#2196f3" backColor="#cacccd" loading={true} />
	</div>
);

export default LoadingPage;
