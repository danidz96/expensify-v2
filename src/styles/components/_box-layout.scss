.box-layout {
	background: linear-gradient(to left, #fc466b, #3f5efb);
	height: 100vh;
	display: flex;
	justify-content: center;
	align-items: center;

	&__box {
		background: white;
		height: 30rem;
		width: 40rem;
		border-radius: $border-1;
		text-align: center;
		padding: $l-size $m-size;
		display: flex;
		flex-direction: column;
		align-items: center;

		button {
			margin-top: auto;
		}
	}

	&__title {
		margin: 0 0 $m-size 0;
		font-size: $font-size-large;
	}
}
