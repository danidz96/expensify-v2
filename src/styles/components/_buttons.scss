.button {
	border-radius: $border;
	padding: $s-size $m-size;
	transition: all 0.2s;
	background-color: $blue;
	border: none;
	color: white;
	text-decoration: none;

	&--action {
		padding: $m-size $l-size;
		border-radius: 5rem;
	}

	&--link {
		background: none;
	}

	&:hover {
		cursor: pointer;
		background-color: $blue-dark;
		transform: translateY(-2px);
		box-shadow: 0 .4rem 2rem #ccc;
		box-shadow: 0px .8rem 1.5rem rgba(0, 0, 0, 0.1);
	}

	&:active {
		outline: none;
		transform: translateY(0);
	}

	&:focus {
		outline: none;
	}
}
