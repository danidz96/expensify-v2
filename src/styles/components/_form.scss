.form {
	display: flex;
	flex-direction: column;
	width: 100%;

	> * {
		margin-bottom: $m-size;
	}

	&__error {
		margin: 0 0 $m-size 0;
	}
}
