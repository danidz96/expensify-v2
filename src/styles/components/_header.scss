.header {
	height: 10rem;
	box-shadow: 0 4px 2px -2px gray;
	background: $indigo;

	&__content {
		display: flex;
		justify-content: space-between;
		align-items: center;
		height: 100%;
	}

	&__title {
		color: white;
		text-decoration: none;

		h1 {
			margin: 0;
		}
	}
}
