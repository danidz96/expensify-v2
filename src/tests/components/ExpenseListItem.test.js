import React from 'react';
import { ExpenseListItem } from '../../components/ExpenseListItem';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';

test('should render item from expense list', () => {
	const wrapper = shallow(<ExpenseListItem {...expenses[1]} />);
	expect(wrapper).toMatchSnapshot();
});
