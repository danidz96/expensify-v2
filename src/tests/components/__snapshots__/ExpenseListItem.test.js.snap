// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render item from expense list 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/2"
  >
    <h3>
      Rent
    </h3>
  </Link>
  <p>
    400$
    -
    28th December, 1969
  </p>
</div>
`;
